public/app/plugins/datasource/loki/components/stats.ts

```diff
@@ -0,0 +1,30 @@
+import { TimeRange } from '@grafana/data';
+
+import { LokiDatasource } from '../datasource';
+import { QueryStats } from '../types';
+
+export async function makeStatsRequest(datasource: LokiDatasource, query: string): Promise<QueryStats | undefined> {
```

Comment:
```suggestion
export async function getStats(datasource: LokiDatasource, query: string): Promise<QueryStats | undefined> {
```

Again, personal preference - but `makeStatsRequest` for me sounds like we are making request and then executing it somewhere else. I would just name this getStats.